librad/src/net/protocol.rs

```diff
@@ -149,6 +150,14 @@ where
 
         tracing::info!(msg = "Listening", local.addr = ?endpoint.local_addr());
 
+        self.subscribers
+            .emit(ProtocolEvent::Listening(
+                endpoint
```

Comment:
Could you maybe bind this at the start of the method, and stick it into the `trace_span!` (like `trace_span!("Protocol::run", local_addr = %local_addr)`). The `info!` can then be removed.

I figure this would much improve log output when there are more than one peers in a single process, as you can tell which one emitted the log line.